Wikipedia/Code/WMFSaveButtonController.h

```diff
@@ -24,6 +24,7 @@
 /**
  *  Set to provide a source for logging saved pages
  */
-@property (weak, nonatomic) id<WMFAnalyticsLogging> analyticsSource;
+@property (weak, nonatomic) id<WMFAnalyticsContextProviding> analyticsContext;
```

Comment:
oh i see your point, it's more important that the context/etc. are all set during configuration time, not at cell creation. just like saved page list :ok_hand: